### PR TITLE
Add back role textbox to the List block and improve aria-multiline usage

### LINKED ISSUE #5983: List block and screen readers: greatly inconsistent behavior
Seems `contenteditable` on a list element doesn't make screen readers so happy.

I've prepared a quick codepen, and I'd greatly appreciate some help and testing: https://codepen.io/afercia/full/aYamBy/

VoiceOver: announces the list but doesn't announce the list items content. It just reads out "group with 2 items editable":

screen shot 2018-04-04 at 19 40 31

NVDA: doesn't read the list item content in forms mode.

Contrary to what I thought, see #4074, seems the only way to address this issue is to reset the list semantics and use a `role="textbox"` for the list. In #4074 we've explicitly excluded lists and tables to get a `role="textbox"`.

Instead, using a `role="textbox"` should, more or less, provide the same experience users get with the classic editor. More testing and research is definitely necessary.

### PR DESCRIPTION
**Pull request by afercia**
## Description
As discussed during WCEU contributor day, the best option for #5983 is to make the list block have the semantics of a textarea. This PR:
- adds back `role=textbox` to the list block editable element
- simplifies the condition, no more reason to use `indexOf()` as we're now checking only against a "table" `tagName`
- also, after #7306 ensures the `aria-multiline` attribute is used only when there's a `role=textbox`. According to the spec, [this attribute can be used only with a textbox role](https://www.w3.org/TR/wai-aria-1.1/#aria-multiline) so it should be omitted for the table element instead of being `false`

@jorgefilipecosta the last point works but maybe can be improved, please do feel free to jump in if you have a chance, thanks! 🙂 

## How has this been tested?
- npm test and updated snapshots

Fixes #5983

### CODE REVIEW

File: editor/components/rich-text/tinymce.js [hunk 1]
```diff
@@ -202,8 +202,12 @@ export default class TinyMCE extends Component {
 	render() {
 		const { tagName = 'div', style, defaultValue, className, isPlaceholderVisible, format } = this.props;
 		const ariaProps = pickAriaProps( this.props );
-		if ( [ 'ul', 'ol', 'table' ].indexOf( tagName ) === -1 ) {
+
+		if ( tagName !== 'table' ) {
 			ariaProps.role = 'textbox';
+		} else {
+			// The `aria-multiline` attribute must be used only together with `role=textbox`.
+			ariaProps[ 'aria-multiline' ] = null;
```
Comment (youknowriad) [inline editor/components/rich-text/tinymce.js]: I think we discussed that we should almost always use `aria-multiline`. So what if we remove it entirely from the `ariaProps` and just add it here where we add `ariaProps.role = 'textbox';`. This seems consistent for me?

Comment (afercia) [reply editor/components/rich-text/tinymce.js]: Yep, we missed the table case... 😬 Still, we should allow to use both `true` (emulating a textarea) and `false` (emulating an input field). Omit it entirely when there's no role=textbox.

Comment (gziolo) [reply editor/components/rich-text/tinymce.js]: Agreed on this one, we should be setting properties rather than removing it 👍

Comment (afercia) [reply editor/components/rich-text/tinymce.js]: Sure, agreed to simplify and always handle aria-multiline internally setting it to `true` as TinyMCE always behaves like a textarea, where text naturally wraps in multiple lines (regardless of the Enter key behavior). Removed the prop from the readme doc.

**Review by youknowriad — APPROVED** [review]
👍 Nice

**Comment by afercia** [comment]
Thanks for the help!

**Pull request merged**